features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/processor/RecordedDataProcessor.kt

```diff
@@ -50,6 +54,25 @@ internal class RecordedDataProcessor(
         )
     }
 
+    @WorkerThread
+    override fun processWebViewRecord(item: WebViewRecordedDataQueueItem) {
+        JsonParser.parseString(item.serializedRecord)?.asJsonObject?.let {
+            val timestampOffset = timeProvider.getTimestampOffset()
+            it.get("timestamp")?.let { timestamp ->
+                val correctedTimestamp = it.asLong + timestampOffset
+                it.addProperty("timestamp", correctedTimestamp)
+            }
```

Comment:
Will `timestamps` be monotonic if changed this way? In WebView → RUM integration, [we do more effort on tracking the "server offset" per "view ID"](https://github.com/DataDog/dd-sdk-android/blob/4744e9d7f85449708c96580ba5f7bd9695dbebc0/features/dd-sdk-android-webview/src/main/kotlin/com/datadog/android/webview/internal/rum/WebViewRumEventConsumer.kt#L92-L100) to avoid skews while NTP sync is pending. I can't see if this problem is somehow addressed by `TimeProvider` or this is sth we should do. WDYT @mariusc83 @0xnm ?

Reply:
what we are using here is the constant timestamp we ware holding in the RumViewScope for that active view and we are dispatching through the RumFeatureContext. I am not sure we should still do what we did for RUM events in this case.